Source/JavaScriptCore/runtime/JSGenericTypedArrayViewPrototype.cpp

```diff
@@ -0,0 +1,94 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "JSGenericTypedArrayViewPrototype.h"
+
+#include "NumberPrototype.h"
+#include <wtf/text/Base64.h>
+
+namespace JSC {
+
+JSC_DEFINE_HOST_FUNCTION(uint8ArrayPrototypeToBase64, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    JSUint8Array* uint8Array = jsDynamicCast<JSUint8Array*>(callFrame->thisValue());
+    if (UNLIKELY(!uint8Array))
+        return throwVMTypeError(globalObject, scope, "Uint8Array.prototype.toBase64 requires that |this| be a Uint8Array"_s);
+
+    validateTypedArray(globalObject, uint8Array);
+
+    Base64EncodeMode mode = Base64EncodeMode::Default;
+
+    JSValue optionsValue = callFrame->argument(0);
+    if (!optionsValue.isUndefined()) {
+        if (!optionsValue.isObject())
+            return throwVMTypeError(globalObject, scope, "Uint8Array.prototype.toBase64 requires that options be an object"_s);
+
+        JSObject* optionsObject = jsCast<JSObject*>(optionsValue);
+
+        JSValue alphabetValue = optionsObject->get(globalObject, vm.propertyNames->alphabet);
+        RETURN_IF_EXCEPTION(scope, { });
+        if (!alphabetValue.isUndefined()) {
+            if (!alphabetValue.isString())
+                return throwVMTypeError(globalObject, scope, "Uint8Array.prototype.toBase64 requires that alphabet be \"base64\" or \"base64url\""_s);
+
+            String alphabet = alphabetValue.toWTFString(globalObject);
+            if (alphabet == "base64url"_s)
+                mode = Base64EncodeMode::URL;
+            else if (alphabet != "base64"_s)
+                return throwVMTypeError(globalObject, scope, "Uint8Array.prototype.toBase64 requires that alphabet be \"base64\" or \"base64url\""_s);
+        }
+    }
+
+    const uint8_t* data = uint8Array->typedVector();
+    size_t length = uint8Array->length();
+    RELEASE_AND_RETURN(scope, JSValue::encode(jsString(vm, base64EncodeToString({ data, length }, mode))));
+}
+
+JSC_DEFINE_HOST_FUNCTION(uint8ArrayPrototypeToHex, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    JSUint8Array* uint8Array = jsDynamicCast<JSUint8Array*>(callFrame->thisValue());
+    if (UNLIKELY(!uint8Array))
+        return throwVMTypeError(globalObject, scope, "Uint8Array.prototype.toHex requires that |this| be a Uint8Array"_s);
+
+    validateTypedArray(globalObject, uint8Array);
+
+    StringBuilder builder;
+    builder.reserveCapacity(uint8Array->length());
+
+    const uint8_t* data = uint8Array->typedVector();
+    size_t length = uint8Array->length();
+    for (size_t i = 0; i < length; ++i)
+        builder.append(toStringWithRadix(data[i], 16));
```

Comment:
This is allocating String per byte, which is too costly.

Reply:
Incase it's helpful, in Bun we implemented the Node.js API `Buffer.prototype.toString("hex")`:

https://github.com/oven-sh/bun/blob/aa3aa888d52b8c7546ac365d56a0cd5c990cb7d8/src/string_immutable.zig#L3949-L4038 

It uses SIMD and is [around 10x faster](https://x.com/jarredsumner/status/1693597759310573945/photo/) than Node.js' scalar implementation